### ISSUE TITLE
kafka plugins fail to install

### ISSUE THREAD

**Issue by colinsurprenant** [elastic/logstash#3123]
in master the kafka plugins installation is failing.

To reproduce:

```sh
rm -rf vendor/bundle .bundle/ Gemfile.jruby-1.9.lock ~/.m2
git checkout Gemfile
rake bootstrap
DEBUG=1 bin/plugin install logstash-input-kafka
```

then, the exception, minus the Bundler traces is:

```sh
[ERROR] Failed to execute goal on project logstash-input-kafka: Could not resolve dependencies for project rubygems:logstash-input-kafka:gem:0.1.8: Failed to collect dependencies at rubygems:logstash:gem:[1.4.0,2.0.0): No versions available for rubygems:logstash:gem:[1.4.0,2.0.0) within specified range -> [Help 1]
[ERROR]
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/DependencyResolutionException

Errno::ENOENT: No such file or directory - /Users/colin/dev/src/elasticsearch/logstash/vendor/bundle/jruby/1.9/gems/logstash-input-kafka-0.1.8/deps.lst
org/jruby/RubyIO.java:3783:in `read'
org/jruby/RubyIO.java:3966:in `read'
/Users/colin/dev/src/elasticsearch/logstash/vendor/jruby/lib/ruby/shared/jar_installer.rb:74:in `load_from_maven'
/Users/colin/dev/src/elasticsearch/logstash/vendor/jruby/lib/ruby/shared/jar_installer.rb:274:in `install_dependencies'
/Users/colin/dev/src/elasticsearch/logstash/vendor/jruby/lib/ruby/shared/jar_installer.rb:204:in `do_install'
/Users/colin/dev/src/elasticsearch/logstash/vendor/jruby/lib/ruby/shared/jar_installer.rb:174:in `vendor_jars'
/Users/colin/dev/src/elasticsearch/logstash/vendor/jruby/lib/ruby/shared/jar_install_post_install_hook.rb:28:in `(root)'
org/jruby/RubyProc.java:271:in `call'
/Users/colin/dev/src/elasticsearch/logstash/vendor/jruby/lib/ruby/shared/rubygems/installer.rb:299:in `run_post_install_hooks'
org/jruby/RubyArray.java:1613:in `each'
/Users/colin/dev/src/elasticsearch/logstash/vendor/jruby/lib/ruby/shared/rubygems/installer.rb:298:in `run_post_install_hooks'
/Users/colin/dev/src/elasticsearch/logstash/vendor/jruby/lib/ruby/shared/rubygems/installer.rb:263:in `install'
/Users/colin/dev/src/elasticsearch/logstash/vendor/bundle/jruby/1.9/gems/bundler-1.9.4/lib/bundler/source/rubygems.rb:114:in `install'
/Users/colin/dev/src/elasticsearch/logstash/vendor/bundle/jruby/1.9/gems/bundler-1.9.4/lib/bundler/rubygems_integration.rb:131:in `preserve_paths'
/Users/colin/dev/src/elasticsearch/logstash/vendor/bundle/jruby/1.9/gems/bundler-1.9.4/lib/bundler/source/rubygems.rb:113:in `install'
/Users/colin/dev/src/elasticsearch/logstash/vendor/bundle/jruby/1.9/gems/bundler-1.9.4/lib/bundler/installer.rb:107:in `install_gem_from_spec'
/Users/colin/dev/src/elasticsearch/logstash/vendor/bundle/jruby/1.9/gems/bundler-1.9.4/lib/bundler/installer.rb:269:in `install_sequentially'
org/jruby/RubyArray.java:1613:in `each'
/Users/colin/dev/src/elasticsearch/logstash/vendor/jruby/lib/ruby/1.9/forwardable.rb:201:in `each'
/Users/colin/dev/src/elasticsearch/logstash/vendor/bundle/jruby/1.9/gems/bundler-1.9.4/lib/bundler/installer.rb:268:in `install_sequentially'
/Users/colin/dev/src/elasticsearch/logstash/vendor/bundle/jruby/1.9/gems/bundler-1.9.4/lib/bundler/installer.rb:90:in `run'
/Users/colin/dev/src/elasticsearch/logstash/vendor/bundle/jruby/1.9/gems/bundler-1.9.4/lib/bundler/installer.rb:18:in `install'
/Users/colin/dev/src/elasticsearch/logstash/vendor/bundle/jruby/1.9/gems/bundler-1.9.4/lib/bundler/cli/install.rb:80:in `run'
/Users/colin/dev/src/elasticsearch/logstash/vendor/bundle/jruby/1.9/gems/bundler-1.9.4/lib/bundler/cli.rb:157:in `install'
/Users/colin/dev/src/elasticsearch/logstash/vendor/bundle/jruby/1.9/gems/bundler-1.9.4/lib/bundler/vendor/thor/lib/thor/command.rb:27:in `run'
/Users/colin/dev/src/elasticsearch/logstash/vendor/bundle/jruby/1.9/gems/bundler-1.9.4/lib/bundler/vendor/thor/lib/thor/invocation.rb:126:in `invoke_command'
/Users/colin/dev/src/elasticsearch/logstash/vendor/bundle/jruby/1.9/gems/bundler-1.9.4/lib/bundler/vendor/thor/lib/thor.rb:359:in `dispatch'
/Users/colin/dev/src/elasticsearch/logstash/vendor/bundle/jruby/1.9/gems/bundler-1.9.4/lib/bundler/vendor/thor/lib/thor/base.rb:440:in `start'
/Users/colin/dev/src/elasticsearch/logstash/vendor/bundle/jruby/1.9/gems/bundler-1.9.4/lib/bundler/cli.rb:10:in `start'
/Users/colin/dev/src/elasticsearch/logstash/lib/logstash/bundler.rb:109:in `invoke_bundler!'
org/jruby/RubyKernel.java:1507:in `loop'
/Users/colin/dev/src/elasticsearch/logstash/lib/logstash/bundler.rb:106:in `invoke_bundler!'
org/jruby/RubyProc.java:271:in `call'
/Users/colin/dev/src/elasticsearch/logstash/lib/logstash/bundler.rb:61:in `capture_stdout'
/Users/colin/dev/src/elasticsearch/logstash/lib/logstash/bundler.rb:105:in `invoke_bundler!'
/Users/colin/dev/src/elasticsearch/logstash/lib/logstash/pluginmanager/install.rb:98:in `install_gems_list!'
/Users/colin/dev/src/elasticsearch/logstash/lib/logstash/pluginmanager/install.rb:34:in `execute'
/Users/colin/dev/src/elasticsearch/logstash/vendor/bundle/jruby/1.9/gems/clamp-0.6.4/lib/clamp/command.rb:67:in `run'
/Users/colin/dev/src/elasticsearch/logstash/vendor/bundle/jruby/1.9/gems/clamp-0.6.4/lib/clamp/subcommand/execution.rb:11:in `execute'
/Users/colin/dev/src/elasticsearch/logstash/vendor/bundle/jruby/1.9/gems/clamp-0.6.4/lib/clamp/command.rb:67:in `run'
/Users/colin/dev/src/elasticsearch/logstash/vendor/bundle/jruby/1.9/gems/clamp-0.6.4/lib/clamp/command.rb:132:in `run'
/Users/colin/dev/src/elasticsearch/logstash/lib/logstash/pluginmanager.rb:12:in `(root)'
```

**Comment by colinsurprenant** [comment]
did not test in 1.5 so I don't know if this is a blocker. can someone else take the lead on this so I can concentrate on #3111 ?

**Comment by suyograo** [comment]
@colinsurprenant I'll check this and work with @talevy

**Comment by suyograo** [comment]
Tried on RC3 binaries. I uninstall kafka (because we bundle it) and install it again. There is an error related to jar dependencies, but it works and gets installed

```
ubuntu@ip-10-0-1-9:~/logstash-1.5.0-rc3$ bin/plugin uninstall logstash-input-kafka
Uninstalling logstash-input-kafka
ubuntu@ip-10-0-1-9:~/logstash-1.5.0-rc3$ find . -name *.jar | grep kafka
ubuntu@ip-10-0-1-9:~/logstash-1.5.0-rc3$ find . -name *.jar | grep kafka
ubuntu@ip-10-0-1-9:~/logstash-1.5.0-rc3$ bin/plugin list logstash-input-kafka
ERROR: No plugins found
ubuntu@ip-10-0-1-9:~/logstash-1.5.0-rc3$ bin/plugin install logstash-input-kafka
Validating logstash-input-kafka
Installing logstash-input-kafka
[ERROR] Failed to execute goal on project jruby-kafka: Could not resolve dependencies for project rubygems:jruby-kafka:gem:1.4.0: Failed to collect dependencies at org.apache.kafka:kafka_2.10:jar:0.8.2.1: Failed to read artifact descriptor for org.apache.kafka:kafka_2.10:jar:0.8.2.1: Could not transfer artifact org.apache.kafka:kafka_2.10:pom:0.8.2.1 from/to central (http://repo.maven.apache.org/maven2): Specified destination directory cannot be created: /home/ubuntu/.m2/repository/org/apache/kafka/kafka_2.10/0.8.2.1 -> [Help 1]
[ERROR]
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/DependencyResolutionException
Error Bundler::InstallError, retrying 1/10
An error occurred while installing jruby-kafka (1.4.0), and Bundler cannot continue.
Make sure that `gem install jruby-kafka -v '1.4.0'` succeeds before bundling.
Error Errno::EBADF, retrying 2/10
Bad file descriptor - Bad file descriptor
Installation successful
```

**Comment by ph** [comment]
@suyograo check if jruby-kafka contains jars?

**Comment by colinsurprenant** [comment]
@suyograo yeah, check if the plugin actually work

**Comment by jsvd** [comment]
In my rc3 setup the install/uninstall was smooth:

```
/tmp/logstash-1.5.0-rc3 % bin/plugin uninstall logstash-input-kafka
Uninstalling logstash-input-kafka
/tmp/logstash-1.5.0-rc3 % bin/plugin install logstash-input-kafka
Validating logstash-input-kafka
Installing logstash-input-kafka
Installation successful
/tmp/logstash-1.5.0-rc3 %
```

**Comment by colinsurprenant** [comment]
@jsvd did you try my reproduction steps?

**Comment by jsvd** [comment]
rc3 release and master after bootstrap + failed plugin install have same version of maven-tools (1.0.7), ruby-maven (3.1.1.0.8) and ruby-maven-libs (3.1.1)

**Comment by jsvd** [comment]
@colinsurprenant yes I confirmed your scenario in master, but apparently works well on rc3

**Comment by suyograo** [comment]
@colinsurprenant @ph yeah the plugin works on RC3 (after uninstall, install)

```
ubuntu@ip-10-0-1-9:~/logstash-1.5.0-rc3$ bin/logstash -e 'input { kafka { topic_id => "test2" } }'
Logstash startup completed
{
       "message" => "hello",
          "tags" => "_jsonparsefailure",
      "@version" => "1",
    "@timestamp" => "2015-04-28T16:30:19.017Z"
}
```

**Comment by talevy** [comment]
I cannot reproduce on master

**Comment by nitish-upadhyay** [comment]
hello everyone, i am getting this following problem. could you help me resolve this?
The error reported is:
  Couldn't find any output plugin named 'kafka'. Are you sure this is correct? Trying to load the kafka output plugin resulted in this error: no such file to load -- jruby-kafka

**Comment by nitish-upadhyay** [comment]
i am using logstash-1.4.2 on centOS.

**Comment by markwalkom** [comment]
@nitish-upadhyay please put this in your own issue, or visit the mailing list - https://groups.google.com/forum/?hl=en-GB#!forum/logstash-users

**Comment by colinsurprenant** [comment]
FYI, the kafka plugins installation is fixed in #3111 by the plugin manager refactor.

**Comment by colinsurprenant** [comment]
closing.
